### ISSUE TITLE
Using generic variable after assigning it in `type` declaration still produces error

### ISSUE THREAD

**Issue by One-Nose** [pylint-dev/pylint#9142]
### Bug description

When declaring a `type` variable, PyLint doesn't recognise the parameter declaration:

```python
type OrInt[T] = T | int
```

Results in:

```
E0601: Using variable 'T' before assignment (used-before-assignment)
```

Possibly related to #9098.

### Command used

```shell
pylint main.py
```

### Pylint output

```python
************* Module main
main.py:1:0: C0114: Missing module docstring (missing-module-docstring)
main.py:1:16: E0601: Using variable 'T' before assignment (used-before-assignment)
```

### Expected behavior

PyLint should know `T` was already declared and not produce an error.

### Pylint version

```shell
pylint 3.0.1
astroid 3.0.0
Python 3.12.0 (tags/v3.12.0:0fb18b0, Oct  2 2023, 13:03:39) [MSC v.1935 64 bit (AMD64)]
```

**Comment by jacobtylerwalls** [comment]
Thanks for the report. Duplicate of #9110. We hope to get a fix out soon.
